specs/NCHitTestKind.md

```diff
@@ -0,0 +1,163 @@
+        mouseData, point));
+        return 0;
+        }
+        
+    }
+// Logic for Forwarding Messages - End
+
+    switch(message) 
+    {
+        ....
+    // Handling this message is important for enabling dragging
+        case WM_NCHITTEST:
+        POINT pt {GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam)};
+        COREWEBVIEW2_HIT_TEST_KIND kind = NULL;
+        m_compositionController->NCHitTestPoint(&pt, &kind);
+
+        if (kind == COREWEBVIEW2_NC_HIT_TEST_KIND::COREWEBVIEW2_NC_HIT_TEST_KIND_HTCAPTION)
+        {
+            return HTCAPTION;
+        }
+    
+        break;
+        ....
+    }
+....
+}
+```
+#### .NET / WinRT
+```c#
+protected override void WndProc(ref Message m)
+{
+    ....
+
+    bool m_isCapturingMouse = false;
+    if (m.Msg == WM_NCHITTEST)
+    {
+        Point p = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+        if (m_compositionController.NCHitTestPoint(ref p) == 
+                ICoreWebview2NCHitTestKind.COREWEBVIEW2_NC_HIT_TEST_KIND_HTCAPTION) 
+        {
+            m.Result = (IntPtr)HTCAPTION;
+        }
+    }
+// Logic for Forwarding Messages - Start
+    else if (m.msg >= WM_MOUSEFIRST && m.msg <= WM_MOUSELAST) 
+    {
+         // isMouseInWebView should be set by your app.
+         if (isMouseInWebView || m.msg == WM_MOUSELEAVE || m_isCapturingMouse) 
+         {
+
+            if (m.msg == WM_NCRBUTTONDOWN && m.w_param == HTCAPTION) 
+            {
+                 // capturing the mouse will allow us to begin forwarding messages 
+                 // to webview
+                     m_isCapturingMouse = true;
+                     SetCapture(mainWindowHwnd);
+            }
+            else if (message ==  WM_RBUTTONUP && GetCapture() == mainWindowHwnd) 
+            {
+                // no longer need to capture the mouse for draggbale regions support 
+                // after this
+                m_isCapturingMouse = false;
+                ReleaseCapture();
+            }
+            
+            // forward mouse messages to WV
+            m_compositionController.SendMouseInput(
+                (COREWEBVIEW2_MOUSE_EVENT_KIND)message,
+                (COREWEBVIEW2_MOUSE_EVENT_VIRTUAL_KEYS)GET_KEYSTATE_WPARAM(wParam),
+                mouseData,
+                point);
+
+            m.Result = 0
+            return;
+         }
+           
+      }
+// Logic for Forwarding Messages - End
+      ....
+}
+       
+```
+# API Details
+## Win32 C++
+```cpp
+typedef enum COREWEBVIEW2_NC_HIT_TEST_KIND {
+  COREWEBVIEW2_NC_HIT_TEST_KIND_HTCAPTION,
+  COREWEBVIEW2_NC_HIT_TEST_KIND_CLIENT,
+  COREWEBVIEW2_NC_HIT_TEST_KIND_NONE // Unsure if this should NONE or OUTOFBOUNDS
+} COREWEBVIEW2_NC_HIT_TEST_KIND;
+
+[uuid(42BF7BA5-917A-4C27-ADA1-EA6969854C16), object, pointer_default(unique)]
+interface ICoreWebView2StagingCompositionController : NULL {
+    [propget] HRESULT NCHitTestPoint([out, retval] COREWEBVIEW2_NC_HIT_TEST_KIND* val);
```

Comment:
I believe this is missing the input `POINT`?

Reply:
Yes, resolved it. Thx